File: AdHoc/config/mySQLConfig.py
from sqlalchemy import create_engine

def db_connect(path):
    HOST = 'quasar-slave-new.c9ajz690mens.us-east-1.rds.amazonaws.com'
    DATABASE = 'quasar'
    login = open(path).read()
    USER_NAME = login.split(':')[0]
    PASSWORD = login.split(':')[1]

    engine = create_engine('mysql+pymysql://' + USER_NAME + ':' + PASSWORD + '@' + HOST + '/' + DATABASE)
    return engine

def run_query(query, credentials):
    import pandas as pd
    engine = db_connect(credentials)

    if '.sql' in query:
        q = open(query, 'r').read()
    else:
        q = query
    df = pd.read_sql_query(q, engine)
    return df

File: AdHoc/config/mySQLConfig.R
library(RMySQL)

quasarConnect <- function(credentials) {
  require(stringr)

  login <- readChar(credentials, file.info(credentials)$size)
  user <- str_split(login, ':')[[1]][1]
  pw <- str_split(login, ':')[[1]][2]

  con <-
    dbConnect(
    MySQL(),
    user=user,
    host='quasar-slave-new.c9ajz690mens.us-east-1.rds.amazonaws.com',
    password=pw,
    dbname='quasar'
    )

  return(con)

}

con <- quasarConnect('~/Documents/quasar_login.txt')
